cli/tsc/dts/lib.deno_net.d.ts

```diff
@@ -223,6 +236,32 @@ declare namespace Deno {
     options: UnixListenOptions & { transport: "unix" },
   ): UnixListener;
 
+  /** Options which can be set when opening a vsock listener via
+   * {@linkcode Deno.listen}.
+   *
+   * @category Network
+   */
+  export interface VsockListenOptions {
+    cid: number;
+    port: number;
+  }
+
+  /** Listen announces on the local transport address.
+   *
+   * ```ts
+   * const listener = Deno.listen({ cid: -1, port: 80, transport: "vsock" })
+   * ```
+   *
+   * Requires `allow-net` permission.
+   *
+   * @tags allow-net
+   * @category Network
+   */
```

Comment:
Ideally has a brief description of vsock. Add that this is an unstable interface.